src/main/java/htsjdk/samtools/util/SortingCollection.java

```diff
@@ -164,7 +167,28 @@ public void add(final T rec) {
             throw new IllegalStateException("Cannot add after calling iterator()");
         }
         if (numRecordsInRam == maxRecordsInRam) {
+            // sample every 100 files written.
+            long startMem = 0;
+            boolean printRecordSizeSampling = files.size() % 100 == 0 && Log.getGlobalLogLevel() == Log.LogLevel.DEBUG;
```

Comment:
Can all this profiling code be activated with a boolean provided on construction?  The default value should be true if debug level is on, and false otherwise. This code performs GC and computes memory usage even if the log is not printed (not debug), and I prefer to being able to switch off and to do not run unless the debug option is specified or I explicitly set it to true...

Reply:
It is set only if your log level is set to DEBUG which, if it is, I assume you are doing some debugging. I could add a flag too but in most cases you'd have to bubble that up in order for it to be useful at all since sorting collection is used under the hood in most places.